data-sources/sourcecred/src/generate-datasets.ts

```diff
@@ -0,0 +1,205 @@
+const sc = require('sourcecred')
+const Passport = require('../abis/Passport.json')
+const Ethers = require('ethers')
+
+/* eslint @typescript-eslint/no-var-requires: "off" */
+const createObjectCsvWriter = require('csv-writer').createObjectCsvWriter
+
+let provider = new Ethers.providers.AnkrProvider()
+let PassportContract = new Ethers.Contract(
+  '0x3337dac9f251d4e403d6030e18e3cfb6a2cb1333',
+  Passport.abi,
+  provider
+)
+
+console.info('PassportContract._address:', PassportContract.address)
+
+const SOURCECRED_URL =
+  'https://raw.githubusercontent.com/nation3/nationcred-instance/gh-pages/'
+
+interface SCAlias {
+  address: string
+  description: string
+}
+
+interface SCIdentity {
+  id: string
+  subtype: string
+  name: string
+  address: string
+  aliases: SCAlias[]
+}
+
+interface CredGrainViewParticipantPlusWallet {
+  active: string
+  identity: SCIdentity
+  cred: number
+  credPerInterval: number[]
+  grainEarned: string
+  grainEarnedPerInterval: string[]
+  walletAddress: string[]
+}
+
+interface Interval {
+  startTimeMs: number
+  endTimeMs: number
+  startDate: string
+  endDate: string
+}
+
+loadSourceCredData()
+
+async function loadSourceCredData() {
+  console.info('loadSourceCredData')
+
+  const instance =
+    sc.sourcecred.instance.readInstance.getNetworkReadInstance(SOURCECRED_URL)
+  const credGrainView = await instance.readCredGrainView()
+  const ledger = await instance.readLedger()
+
+  console.info(`Loaded N3 SourceCred instance`)
+
+  let intervalsWeCareAbout: Array<[Interval, number]> =
+    buildIntervals(credGrainView)
+
+  const peopleWhoDidtuffMap = getParticipantsWhoParticipated(
+    ledger,
+    credGrainView
+  )
+
+  const nextId: number = await getNextId()
+  console.info('nextId:', nextId)
+
+  let passportId: number
+  for (passportId = 0; passportId < nextId; passportId++) {
+    console.info('passportId:', passportId)
+
+    const signerAddress: string = await getSigner(passportId)
+    console.info('signerAddress:', signerAddress)
+
+    const writer = createObjectCsvWriter({
+      path: `output/sourcecred-${signerAddress}.csv`,
+      header: [
+        { id: 'week_end', title: 'week_end' },
+        { id: 'sourcecred_score', title: 'sourcecred_score' },
+      ],
+    })
+    const csvRows: any[] = []
+
+    let participant: CredGrainViewParticipantPlusWallet[] | undefined =
+      peopleWhoDidtuffMap.get(signerAddress)
+
+    console.info(
+      `${passportId} linked to wallet? ${participant ? ' - YES' : ' - NO'}`
+    )
+
+    intervalsWeCareAbout.forEach((interval) => {
+      const csvRow = {
+        week_end: interval[0].endDate,
+        sourcecred_score: participant
+          ? sumCredForIdentities(interval, participant)
+          : 0,

```

Comment:
It would have been easier :D If we want it, it would be trivial to add *or* we could create multiple files keyed of wallet address && identity (but identity names would need parsing as they are long and full of #####)